homeassistant/components/honeywell/__init__.py

```diff
@@ -1 +1,147 @@
 """Support for Honeywell (US) Total Connect Comfort climate systems."""
+from datetime import timedelta
+import logging
+
+import requests
+import somecomfort
+import voluptuous as vol
+
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import load_platform
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.util import Throttle
+
+from .const import (
+    CONF_COOL_AWAY_TEMPERATURE,
+    CONF_DEV_ID,
+    CONF_HEAT_AWAY_TEMPERATURE,
+    CONF_LOC_ID,
+    DEFAULT_COOL_AWAY_TEMPERATURE,
+    DEFAULT_HEAT_AWAY_TEMPERATURE,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=60)
+
+CONFIG_SCHEMA = vol.Schema(
```

Comment:
For integrations that connect to devices or services, we no longer accept new YAML configuration or changes.

This integration needs to be refactored first to use a configuration flow and config entries first.

More information about this can be found in Architecture Decision Record:

- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

See our developer documentation on how to get started creating a configuration flow for this integration:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>

We (and our community!) would really appreciate it if you could start on the refactoring of this integration in a new PR.

Thanks already! :+1:

Reply:
I've added a new config flow for this integration now. Has full test coverage:
```
----------- coverage: platform linux, python 3.8.9-final-0 -----------
Name                                                Stmts   Miss  Cover   Missing
---------------------------------------------------------------------------------
homeassistant/components/honeywell/__init__.py         69     51    26%   51-81, 89-93, 101-122, 127-144
homeassistant/components/honeywell/climate.py         190     68    64%   90-100, 132, 143, 155-160, 165, 170-174, 179-183, 193, 203, 208-210, 224, 229-231, 236-238, 243, 248, 253, 263, 266-268, 274, 281-287, 290-291, 302, 305-307, 311, 328-330, 339-340, 351-352, 356-359, 363, 367-370
homeassistant/components/honeywell/config_flow.py      22      0   100%
homeassistant/components/honeywell/const.py             7      0   100%
---------------------------------------------------------------------------------
TOTAL                                                 288    119    59%
```